packages/core-contracts/contracts/utils/AddressSet.sol

```diff
@@ -0,0 +1,70 @@
+//SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+library AddressSet {
```

Comment:
Any idea on what will our strategy be when we need to support other types? Will we duplicate code?

Reply:
We could duplicate code, or just bring the entire util from OZ's code which is compatible with multiple types: https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/utils/structs/EnumerableSet.sol

I didn't do that last option just because we were needing it at the moment :)

Reply:
Yeah, A generic enumerable set might not be very gas efficient, haven't checked tho.

Reply:
For now, I'm going to remove the Set entirely and refactor it when needing it.